girder/utility/filesystem_assetstore_adapter.py

```diff
@@ -303,7 +316,12 @@ def deleteFile(self, file):
         if matching.count(True) == 1:
             path = os.path.join(self.assetstore['root'], file['path'])
             if os.path.isfile(path):
-                os.remove(path)
+                with filelock.FileLock(path + '.deleteLock'):
```

Comment:
I think we may need to move this lock acquisition up above the file collection query -- there's another race condition here. If we do the file query, but someone uploads a duplicate after the file query returns 1 result, we'd delete the file erroneously.